### PR TITLE
fixes issue of only iterating a single collection before starting tail

### PR DESCRIPTION
**Pull request by jipperinbham**
the current code and tests for tailer were not taking into account multiple collections, when the updated test was executed with the existing `experimental` branch, the following errors occurred:

```
=== RUN   TestTail
INFO[0018] setting oplog start timestamp: 6376246771911229440
INFO[0018] starting Read func                            db="tail_test"
INFO[0018] collection count                              db="tail_test" num_collections=4
INFO[0018] sending for iteration...                      collection=blah db="tail_test"
INFO[0018] sending for iteration...                      collection=boo db="tail_test"
INFO[0018] iterating...                                  collection=blah
INFO[0018] Read completed                                db="tail_test"
INFO[0018] tailing oplog with query map[ts:map[$gte:6376246771911229440]]
  db="tail_test"
INFO[0023] tailing oplog with query map[ts:map[$gte:6376246771911229440]]
  db="tail_test"
--- FAIL: TestTail (12.52s)
	tailer_test.go:118: [initial drain] bad message count, expected 150, got 50
```

After the updates to `pkg/adaptor/mongodb/reader.go`, the updated tests pass.

### CODE REVIEW

File: pkg/adaptor/mongodb/reader.go [hunk 1]
```diff
@@ -85,6 +85,7 @@ func (r *Reader) listCollections(mgoSession *mgo.Session, filterFn func(name str
 			}
 		}
 		log.With("db", r.db).Infoln("done iterating collections")
+		return
```
Comment (trinchan) [inline pkg/adaptor/mongodb/reader.go]: I'm not sure why this is necessary, it's right before exiting, no?

File: pkg/adaptor/mongodb/reader.go [hunk 2]
```diff
@@ -95,7 +96,10 @@ func (r *Reader) iterateCollection(mgoSession *mgo.Session, in <-chan string, do
...
 		for {
 			select {
-			case c := <-in:
+			case c, ok := <-in:
+				if !ok {
+					return
+				}
 				log.With("collection", c).Infoln("iterating...")
 				canReissueQuery := r.requeryable(c, mgoSession)
 				var lastID interface{}
```
Comment (trinchan) [inline pkg/adaptor/mongodb/reader.go]: Change above looks OK.

Looking at `lastID`, I don't see where we're updating it. I think it needs to be updated in the `iter.Next` loop below. Wonder if it's possible to write a test that captures the requerying logic.

File: pkg/adaptor/mongodb/reader.go [hunk 3]
```diff
@@ -119,7 +123,7 @@ func (r *Reader) iterateCollection(mgoSession *mgo.Session, in <-chan string, do
 					}
 					iter.Close()
 					s.Close()
-					return
+					break
```
Comment (trinchan) [inline pkg/adaptor/mongodb/reader.go]: So in this case, the query didn't error. We break and wait for a new collection to iterate or the channel to close.. that seems correct. Good catch.

File: pkg/adaptor/mongodb/reader.go (hunk 2 above)
Comment (jipperinbham) [inline pkg/adaptor/mongodb/reader.go]: good catch, looks like setting `lastID` got lost in the refactor. I'll get another PR together for that.

File: pkg/adaptor/mongodb/reader.go (hunk 3 above)
Comment (jipperinbham) [inline pkg/adaptor/mongodb/reader.go]: correct

File: pkg/adaptor/mongodb/reader.go (hunk 2 above)
Comment (jipperinbham) [inline pkg/adaptor/mongodb/reader.go]: I went ahead and put the old code back and will look at adding a test for it as well.

**Pull request merged**